webapp/src/components/App/Singlestudy/explore/Configuration/General/dialogs/ScenarioPlaylistDialog/index.tsx

```diff
@@ -0,0 +1,242 @@
+ *
+ * This file is part of the Antares project.
+ */
+
+import CustomScrollbar from "@/components/common/CustomScrollbar";
+import DataGridForm, {
+  type DataGridFormApi,
+  type DataGridFormProps,
+  type DataGridFormState,
+} from "@/components/common/DataGridForm";
+import ConfirmationDialog from "@/components/common/dialogs/ConfirmationDialog";
+import useConfirm from "@/hooks/useConfirm";
+import { getPlaylistData, setPlaylistData } from "@/services/api/studies/config/playlist";
+import { DEFAULT_WEIGHT } from "@/services/api/studies/config/playlist/constants";
+import type { Playlist, PlaylistData } from "@/services/api/studies/config/playlist/types";
+import { translateWithColon } from "@/utils/i18nUtils";
+import { Box, Button, ButtonGroup } from "@mui/material";
+import * as R from "ramda";
+import * as RA from "ramda-adjunct";
+import { useMemo, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useUpdateEffect } from "react-use";
+import usePromise from "../../../../../../../../hooks/usePromise";
+import type { StudyMetadata } from "../../../../../../../../types/types";
+import BasicDialog from "../../../../../../../common/dialogs/BasicDialog";
+import type { SubmitHandlerPlus } from "../../../../../../../common/Form/types";
+import UsePromiseCond from "../../../../../../../common/utils/UsePromiseCond";
+import ScenariosSelectionFE from "./YearsSelectionFE";
+interface Props {
+  study: StudyMetadata;
+  open: boolean;
+  onClose: VoidFunction;
+}
+
+function ScenarioPlaylistDialog({ study, open, onClose }: Props) {
+  const { t } = useTranslation();
+  const dataGridApiRef = useRef<DataGridFormApi<PlaylistData>>(null);
+  const [yearsSelection, setYearsSelection] = useState<number[] | null>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isDirty, setIsDirty] = useState(false);
+  const [totals, setTotals] = useState({ selected: 0, sumWeights: 0 });
+  const closeAction = useConfirm();
+  const res = usePromise(() => getPlaylistData({ studyId: study.id }), [study.id]);
+  const disableBtnGroup = isSubmitting || (yearsSelection !== null && yearsSelection.length === 0);
+
+  const columns = useMemo<DataGridFormProps<PlaylistData>["columns"]>(() => {
+    return [
+      {
+        id: "status" as const,
+        title: t("global.status"),
+        grow: 1,
+        trailingRowOptions: {
+          hint: `${translateWithColon(t("study.configuration.general.mcScenarioPlaylist.totalSelected.label"))} ${totals.selected}`,
+        },
+      },
+      {
+        id: "weight" as const,
+        title: t("global.weight"),
+        grow: 1,
+        trailingRowOptions: {
+          hint: `${translateWithColon(t("study.configuration.general.mcScenarioPlaylist.totalWeight.label"))} ${totals.sumWeights}`,
+        },
+      },
+    ];
+  }, [t, totals]);
+
+  useUpdateEffect(() => {
+    if (res.data) {
+      updateTotals(res.data);
+    }
+  }, [res.data]);
```

Comment:
It's called both the `useUpdateEffect` and via the `onDataChange` prop of `DataGridForm`, the totals might be updated twice in some scenarios.